x/tbr/internal/keeper/keeper.go

```diff
@@ -0,0 +1,191 @@
+package keeper
+
+import (
+	"github.com/commercionetwork/commercionetwork/x/tbr/internal/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/cosmos/cosmos-sdk/x/distribution"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/cosmos/cosmos-sdk/x/staking/exported"
+)
+
+type Keeper struct {
+	StoreKey sdk.StoreKey
+
+	BankKeeper         bank.Keeper
+	StakeKeeper        staking.Keeper
+	DistributionKeeper distribution.Keeper
+
+	Cdc *codec.Codec
+}
+
+func NewKeeper(storeKey sdk.StoreKey, bk bank.Keeper, sk staking.Keeper, dk distribution.Keeper, cdc *codec.Codec) Keeper {
+	return Keeper{
+		StoreKey:           storeKey,
+		BankKeeper:         bk,
+		StakeKeeper:        sk,
+		DistributionKeeper: dk,
+		Cdc:                cdc,
+	}
+}
+
+//Utility method to set Block Reward Pool
+func (k Keeper) SetBlockRewardsPool(ctx sdk.Context, updatedPool sdk.DecCoins) {
+	store := ctx.KVStore(k.StoreKey)
+	store.Set([]byte(types.BlockRewardsPoolPrefix), k.Cdc.MustMarshalBinaryBare(&updatedPool))
+}
+
+//Return the Block Rewards Pool if exists
+func (k Keeper) GetBlockRewardsPool(ctx sdk.Context) sdk.DecCoins {
+	var brPool sdk.DecCoins
+	store := ctx.KVStore(k.StoreKey)
+	brpBz := store.Get([]byte(types.BlockRewardsPoolPrefix))
+	k.Cdc.MustUnmarshalBinaryBare(brpBz, &brPool)
+	return brPool
+}
+
+//Increase the Block Rewards Pool with the specified coin amount
+func (k Keeper) IncrementBlockRewardsPool(ctx sdk.Context, funder sdk.AccAddress, amount sdk.Coins) error {
+	bk := k.BankKeeper
+	brPool := sdk.DecCoins{}
+
+	if bk.HasCoins(ctx, funder, amount) {
+		brPool = k.GetBlockRewardsPool(ctx)
+		_, err := bk.SubtractCoins(ctx, funder, amount)
+		if err != nil {
+			return err
+		}
+		brPool = brPool.Add(sdk.NewDecCoins(amount))
+		k.SetBlockRewardsPool(ctx, brPool)
+	}
+	return sdk.ErrUnauthorized("Not enough tokens to supply the given funds' amount")
```

Comment:
I think this should be changed in order to use the `errors` type

```suggestion
	return errors.New("funder does not have enough tokens")
```